src/ArrayValueCalculator.java

```diff
@@ -0,0 +1,56 @@
+package src;
+
+public class ArrayValueCalculator {
+    public static void main(String[] args) throws Exception {
+        try {
+            System.out.println(doCalc(new String[][]{{"5", "22", "44", "12212"}, {"1", "22", "1", "213"}, {"1", "2", "1", "3434"}, {"1", "1", "1", "434"}}));
+        } catch (ArraySizeException e) {
+            System.out.println(e.getMessage());
+        } catch (ArrayDataException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public static int doCalc(String arr[][]) throws ArrayDataException, ArraySizeException {
+        int sum = 0;
+
+        for (int i = 0; i < arr.length; i++) {
+            if ((arr.length | arr[i].length) != 4) {
```

Comment:
У массива всегда есть длина. Не стоит усложнять.
```suggestion
            if (arr[i].length != 4) {
```